playback/src/player.rs

```diff
@@ -771,27 +790,59 @@ impl PlayerState {
     }
 }
 
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub struct MetaData {
+    pub track_id: SpotifyId,
+    pub item_type: SpotifyItemType,
```

Comment:
> I'm very happy with the current solution, just two last comments from my side.
Thank you for bearing with me and my feedback through this PR, hope I haven't been too pedantic!

Don't be sorry I appreciate everything you've said. This is primarily for you so your input is highly valued.

I think you're going in the right direction. I will see what I can come up with after work tonight.

On a side note I wonder if we can't shrink and flatten Track and Episode a bit. There's a lot there that isn't even populated with actual real data. But I'm not sure if it eventually will be once dev finally transitioned over to the new API. As it sits both are extremely bloated.

Reply:
Just look at the podcast message above and compare it to Episode. Track is much worse.

Reply:
> On a side note I wonder if we can't shrink and flatten Track and Episode a bit. There's a lot there that isn't even populated with actual real data. But I'm not sure if it eventually will be once dev finally transitioned over to the new API. As it sits both are extremely bloated.

Yeah, I have been thinking about that too. I might create a new discussion later, since there have been talks about this in several other places too, and it might be a good idea to create a common place for them.

Reply:
We have no way of knowing whether Spotify will ever populate those fields. They are still in the protobuf definitions anyway... I also have the impression that the same protobufs are used for some producer frontend to manage and curate the music library. So it may well be that Spotify has loads more in the database than it currently exposes.

Reply:
Every once in a while you could check the messages? In the meantime they could be commented out and added back if and when they show up?

Reply:
You could easily make them 1/2 the size or less.